test/python/rest/submit_tests.py

```diff
@@ -0,0 +1,212 @@
+
+
+    def test_A_submit_with_defaults(self):
+
+        logger.warning('submit bundle ' + os.path.basename(applicationBundle))
+        result = self.service.submit_job(applicationBundle)
+        self.assertFalse('status_code' in result, 'submit_job() failed, status code and error description:' + str(result))
+        self.assertTrue('name' in result, 'submit_job() failed, no job name returned, result: ' + str(result))
+
+        logger.warning('let the job run a while')
+        time.sleep(runInterval)
+
+        logger.warning('check job ' + result['name'])
+        jobs = self.instance.get_jobs(result['name'])
+        self.assertTrue(len(jobs)>0, 'submit_job() failed, could not find job')
+
+        for job in jobs:
+            self.assertTrue(result['name']==job.name, 'submit_job() failed, could not find job')
+            self.assertTrue(job.status=='running', 'submit_job() failed, job is not running, status is ' + job.status + ', health is ' + job.health)
+            self.assertTrue(job.health=='healthy', 'submit_job() failed, job is not healthy, status is ' + job.status + ', health is ' + job.health)
+
+        for job in jobs:
+            logger.warning('store logs for job ' + job.name)
+            filename = job.get_application_logs()
+            self.assertTrue(os.path.isfile(filename), 'get_application_logs() failed, no file stored as ' + filename)
+            #filename = job.store_logs()
+            #self.assertTrue(os.path.isfile(filename), 'store_logs() failed, no file stored as ' + filename)
+            #for pe in job.get_pes():
+            #    logger.warning('store logs for PE ' + pe.id + ' of job ' + job.name)
+            #    filename = pe.store_console_log()
+            #    self.assertTrue(os.path.isfile(filename), 'store_console_log() failed, no file stored as ' + filename)
+            #    filename = pe.store_application_trace()
+            #    self.assertTrue(os.path.isfile(filename), 'store_application_trace() failed, no file stored as ' + filename)
+
+        for job in jobs:
+            logger.warning('cancel job ' + job.name)
+            result = job.cancel()
+            self.assertTrue(result, 'cancel() failed for job ' + job.name)
+
+
+    def test_B_submit_fully_fused(self):
+
+        jobConfig = streamsx.topology.context.JobConfig(target_pe_count=1, tracing='info')
+
+        jobConfigOverlay = {}
+        jobConfig._add_overlays(jobConfigOverlay)
+
+        logger.warning('submit bundle ' + os.path.basename(applicationBundle))
+        result = self.service.submit_job(applicationBundle, configuration=jobConfigOverlay)
+        self.assertFalse('status_code' in result, 'submit_job() failed, status code and error description:' + str(result))
+        self.assertTrue('name' in result, 'submit_job() failed, no job name returned, result: ' + str(result))
+
+        logger.warning('let the job run a while')
```

Comment:
Ideally use of the existing test framework would simplify this test, as it handles waiting for the job to become healthy, cancelling it, getting the rest connection etc. Though I'm not sure it can handle submission from a sab yet. More a note to myself to improve the test harness to support testing a bundle.